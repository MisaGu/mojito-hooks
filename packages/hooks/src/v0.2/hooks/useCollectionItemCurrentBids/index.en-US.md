---
nav:
  path: /hooks
  group: Mojito
---

# useCollectionItemCurrentBids

This slugs loads all collection items from the selected collection, including its current bids, using
`useCollectionItemsCurrentBids(...)`, and returns only the selected item (using the `collectionItemID` param).

## Examples

### Default usage

<code src="./demo/demo1.tsx" />
