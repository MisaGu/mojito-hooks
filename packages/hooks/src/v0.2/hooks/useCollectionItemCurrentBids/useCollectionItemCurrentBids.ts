import { IMojitoCollectionItemCurrentBids } from '../../domain/interfaces';
import { QueryResult } from '../../domain/utils/gql.utils';
import { BaseHookPropsWithUrlAndSlug } from '../../domain/interfaces/hooks.interface';
import { useCollectionItemsCurrentBids } from '../useCollectionItemsCurrentBids/useCollectionItemsCurrentBids';

export type UseCollectionItemCurrentBidsData = undefined | IMojitoCollectionItemCurrentBids;

export type UseCollectionItemCurrentBidsReturn = QueryResult<
  'itemCurrentBids',
  UseCollectionItemCurrentBidsData
>;

export interface UseCollectionItemCurrentBidsProps
  extends BaseHookPropsWithUrlAndSlug<UseCollectionItemCurrentBidsData> {
  collectionItemID: string;
}

export function useCollectionItemCurrentBids({
  collectionItemID,
  ...props
}: UseCollectionItemCurrentBidsProps): UseCollectionItemCurrentBidsReturn {
  const { currentBids, ...result } = useCollectionItemsCurrentBids(props as any);

  const itemCurrentBids = currentBids?.find((item) => item.id === collectionItemID);

  return {
    ...(result as any),
    itemCurrentBids,
  };
}
