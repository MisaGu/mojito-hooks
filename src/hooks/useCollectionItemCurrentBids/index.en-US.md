---
nav:
  path: /hooks
  group: Mojito
---

# useCollectionItemCurrentBids

This slugs loads all collection items from the selected collection, including its current bids, using
`useCollectionItemsCurrentBids(...)`, and returns only the selected item (using the `collectionItemID` param).

## Examples

<code src="./demo/demo1.tsx" />
