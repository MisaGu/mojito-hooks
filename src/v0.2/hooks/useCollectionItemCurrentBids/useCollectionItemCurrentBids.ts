import { useMemo } from 'react';
import { MojitoMarketplaceAuctionBid } from '../../domain/interfaces';
import { BaseQueryHookPropsWithUrlAndSlug } from '../../domain/interfaces/hooks.interface';
import { useCollectionItemsCurrentBids } from '../useCollectionItemsCurrentBids/useCollectionItemsCurrentBids';

export type UseCollectionItemCurrentBidsData = undefined | MojitoMarketplaceAuctionBid;

export type UseCollectionItemCurrentBidsReturn = ReturnType<typeof useCollectionItemCurrentBid>;

export interface UseCollectionItemCurrentBidsProps
  extends BaseQueryHookPropsWithUrlAndSlug<UseCollectionItemCurrentBidsData> {
  collectionItemID: string;
}

export function useCollectionItemCurrentBid({
  collectionItemID,
  ...props
}: UseCollectionItemCurrentBidsProps) {
  const { currentBids, ...result } = useCollectionItemsCurrentBids(props as any);
}

export default useCollectionItemCurrentBid;
