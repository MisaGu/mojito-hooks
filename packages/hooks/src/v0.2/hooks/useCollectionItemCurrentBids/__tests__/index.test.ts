import { useCollectionItemCurrentBids } from '../useCollectionItemCurrentBids';

describe('useCollectionItemCurrentBids', () => {
  it('should be defined', () => {
    expect(useCollectionItemCurrentBids).toBeDefined();
  });
});
