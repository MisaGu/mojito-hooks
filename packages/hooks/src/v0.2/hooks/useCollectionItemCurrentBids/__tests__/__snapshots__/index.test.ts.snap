// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useCollectionItemCurrentBids() returns the right result when needed 1`] = `
Object {
  "error": null,
  "isLoading": false,
  "itemCurrentBids": undefined,
  "queryResult": Object {
    "dataUpdatedAt": Any<Number>,
    "errorUpdateCount": 0,
    "errorUpdatedAt": 0,
    "failureCount": 0,
    "isError": false,
    "isFetched": true,
    "isFetchedAfterMount": true,
    "isFetching": false,
    "isIdle": false,
    "isLoadingError": false,
    "isPlaceholderData": false,
    "isPreviousData": false,
    "isRefetchError": false,
    "isRefetching": false,
    "isStale": false,
    "isSuccess": true,
    "remove": [Function],
    "status": "success",
  },
  "refetch": [Function],
}
`;
